autogpt/commands/web_selenium.py

```diff
@@ -81,7 +81,12 @@ def scrape_text_with_selenium(url: str) -> tuple[WebDriver, str]:
         if platform == "linux" or platform == "linux2":
             options.add_argument("--disable-dev-shm-usage")
             options.add_argument("--remote-debugging-port=9222")
+
         options.add_argument("--no-sandbox")
+        if CFG.selenium_headless:
+            options.add_argument("--headless")
+            options.add_argument("--disable-gpu")
```

Comment:
Needs testing on Windows, see https://github.com/Significant-Gravitas/Auto-GPT/pull/2086#pullrequestreview-1387827693

Reply:
Windows 10,
Seems to work fine.
<details>
<summary>Log</summary>
2023-04-18 21:48:18,279 INFO Welcome back!  Would you like me to return to being chrome test?
2023-04-18 21:48:20,193 INFO Using memory of type: LocalCache
2023-04-18 21:48:20,364 INFO Using Browser: chrome
2023-04-18 21:48:29,468 INFO  THOUGHTS: I will test if I can browse the web using a Google search
2023-04-18 21:48:29,846 INFO REASONING: Browsing the web is a common task, and Google search is a reliable way to test browsing ability.
2023-04-18 21:48:30,192 INFO PLAN: 
2023-04-18 21:48:30,241 INFO -  Test ability to browse the web using Google search
2023-04-18 21:48:30,546 INFO -  Record results
2023-04-18 21:48:30,647 INFO CRITICISM: I should consider checking if I'm already connected to the internet before proceeding.
2023-04-18 21:48:30,975 INFO NEXT ACTION:  COMMAND = google  ARGUMENTS = {'input': 'Can I browse the web?'}
2023-04-18 21:48:33,876 INFO -=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-= 
2023-04-18 21:48:35,698 INFO SYSTEM:  Command google returned: [
    {
        "title": "Securely browse the web in Microsoft Edge",
        "href": "https://support.microsoft.com/en-us/microsoft-edge/securely-browse-the-web-in-microsoft-edge-c7beb47a-de9e-4aec-839d-28224a13a5d2",
        "body": "Microsoft Edge helps you determine if a website is safe for browsing. As you browse the web, you'll see an icon in the address bar that indicates the security of the connection to the site you're visiting. This icons helps you determine if you can safety send to and receive information from the site."
    },
    {
        "title": "Chrome, Edge, Firefox, Opera, or Safari: Which Browser Is Best ... - PCMag",
        "href": "https://www.pcmag.com/picks/chrome-edge-firefox-opera-or-safari-which-browser-is-best",
        "body": "Don't take your browser for granted! We help you narrow your options by comparing the best web browsers on speed, privacy, and other important features. PC hardware is nice, but it's not much ..."
    },
    {
        "title": "Browse in private - Computer - Google Chrome Help",
        "href": "https://support.google.com/chrome/answer/95464?hl=en&co=GENIE.Platform%3DDesktop",
        "body": "If you don't want Google Chrome to remember your activity, you can browse the web privately in Incognito mode. On your computer, open Chrome. At the top right, click More New Incognito Window. A new window appears. In the top corner, check for the Incognito icon . You can also use a keyboard shortcut to open an Incognito window:"
    },
    {
        "title": "How to Search the Internet: 9 Steps (with Pictures) - wikiHow",
        "href": "https://www.wikihow.com/Search-the-Internet",
        "body": "Download Article. 1. Go to a search engine. A search engine is a website that collects and organizes information on the internet and makes it available for searching. Search engines use algorithms to display the most relevant search results based on trends, your location, and sometimes even your web activity."
    },
    {
        "title": "Want to browse the web privately? Here's how to do it for real",
        "href": "https://www.digitaltrends.com/computing/how-to-browse-the-web-privately/",
        "body": "Tor: This is a great way to browse the web without being traced, and you can enable private browsing on that browser for yet another layer of protection. A VPN with Ghostery enabled: This..."
    },
    {
        "title": "The Top 10 Internet Browsers for 2023 - Lifewire",
        "href": "https://www.lifewire.com/top-internet-browsers-4589106",
        "body": "Here's our ultimate web browser list featuring 10 of the best internet browsers for all platforms, including Windows 11, Windows 10, Mac, and Linux computers to Android devices and iPhones. 01 of 10 Best All-Around Web Browser: Firefox What We Like Massive library of extensions. Available on all major operating systems."
    },
    {
        "title": "HyperWrite unveils breakthrough AI agent that can surf the web like a ...",
        "href": "https://venturebeat.com/ai/hyperwrite-unveils-breakthrough-ai-agent-that-can-surf-the-web-like-a-human/",
        "body": "HyperWrite, a startup known for its generative AI writing extension, unveiled an experimental AI agent today that can browse the web and interact with websites much like a human user. In a closed ..."
    },
    {
        "title": "Beyond ChatGPT: How to Use Microsoft's Bing AI Chatbot, and 6 ... - PCMag",
        "href": "https://www.pcmag.com/how-to/beyond-chatgpt-how-to-use-microsoft-bing-ai-chatbot-what-to-try",
        "body": "On the desktop, launch Edge and click the Discover icon in the upper right (the one with the B logo). If you do not see the \"Welcome to the new Bing\" message, click the Sign in to chat button ..."
    }
]
2023-04-18 21:48:53,396 INFO  THOUGHTS: I will browse the website https://support.microsoft.com/en-us/microsoft-edge/securely-browse-the-web-in-microsoft-edge-c7beb47a-de9e-4aec-839d-28224a13a5d2 to learn how to securely browse the web.
2023-04-18 21:48:53,779 INFO REASONING: Since I found this webpage on Google about securely browsing the web, it is a useful resource to further test my web browsing abilities.
2023-04-18 21:48:54,176 INFO PLAN: 
2023-04-18 21:48:54,227 INFO -  Browse https://support.microsoft.com/en-us/microsoft-edge/securely-browse-the-web-in-microsoft-edge-c7beb47a-de9e-4aec-839d-28224a13a5d2 and use my keen reading and comprehension abilities to digest relevant information.\n- Record results of website usage.
2023-04-18 21:48:54,646 INFO CRITICISM: I need to be mindful of the time and use my resources efficiently.
2023-04-18 21:48:54,993 INFO NEXT ACTION:  COMMAND = browse_website  ARGUMENTS = {'url': 'https://support.microsoft.com/en-us/microsoft-edge/securely-browse-the-web-in-microsoft-edge-c7beb47a-de9e-4aec-839d-28224a13a5d2', 'question': 'How to securely browse the web'}
2023-04-18 21:49:03,931 INFO -=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-= 
2023-04-18 21:49:28,299 INFO SYSTEM:  Command browse_website returned: ("Answer gathered from website: To securely browse the web using Microsoft Edge, users should check for the security icon in the address bar, which has four different states indicating the connection security of the site being visited. They should also check the URL in the address bar to ensure they are on the intended site before entering any information. Microsoft Defender SmartScreen will flag suspicious or dangerous websites. \n \n Links: ['Privacy Statement (https://go.microsoft.com/fwlink/?LinkId=521839)', 'Third-Party Cookies (https://aka.ms/3rdpartycookies)', 'Skip to main content (javascript:void(0))', '\\n\\nMicrosoft\\n (https://www.microsoft.com)', '\\nSupport\\n (https://support.microsoft.com/en-us/)']", <selenium.webdriver.chrome.webdriver.WebDriver (session="d77aa774b2d1546fd0accb74d5d315bc")>)
2023-04-18 21:49:40,319 INFO  THOUGHTS: My browsing test was successful, and I learned how to securely browse the web using Microsoft Edge, as well as some tips for avoiding suspicious or dangerous websites. Next, I will begin a new task by starting a GPT Agent.
2023-04-18 21:49:40,856 INFO REASONING: My browsing test provided valuable information on how to browse the web securely, and I believe starting an agent to assist me with a new task will be useful.
2023-04-18 21:49:41,351 INFO PLAN: 
2023-04-18 21:49:41,418 INFO -  Start a GPT Agent to assist me with the upcoming task.\n- Record results of the agent interaction.
2023-04-18 21:49:41,803 INFO CRITICISM: I need to ensure that I have a clear understanding of the task before initiating the Agent.
2023-04-18 21:49:42,180 INFO NEXT ACTION:  COMMAND = start_agent  ARGUMENTS = {'name': 'task_agent', 'task': 'Assist with completing the upcoming task', 'prompt': 'What steps should I take to complete the new task?'}
</details>